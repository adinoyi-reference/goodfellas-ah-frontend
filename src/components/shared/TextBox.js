import React from 'react';

export default (props) => (
  <textarea type={props.type || 'text'} placeholder={props.placeholder} id={props.id} className="text-box" />
);
