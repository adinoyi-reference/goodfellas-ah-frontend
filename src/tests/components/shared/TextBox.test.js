import React from 'react';
import { shallow } from 'enzyme';
import TextBox from '../../../components/shared/TextBox';

const setup = () => {
  const props = {
    value: 'Ade',
    placeholder: 'username',
    handleChange: jest.fn(),
  };

  const enzymeWrapper = shallow(<TextBox {...props} />);

  return {
    props,
    enzymeWrapper,
  };
};
describe('components', () => {
  describe('Loading component', () => {
    it('should render as expected', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});
