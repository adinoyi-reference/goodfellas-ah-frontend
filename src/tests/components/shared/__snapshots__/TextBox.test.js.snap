// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Loading component should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    handleChange={[MockFunction]}
    placeholder="username"
    value="Ade"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "text-box",
      "name": undefined,
      "onChange": [MockFunction],
      "placeholder": "username",
      "type": "text",
      "value": "Ade",
    },
    "ref": null,
    "rendered": null,
    "type": "textarea",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "text-box",
        "name": undefined,
        "onChange": [MockFunction],
        "placeholder": "username",
        "type": "text",
        "value": "Ade",
      },
      "ref": null,
      "rendered": null,
      "type": "textarea",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
